### PR TITLE
[WiP] AMQP PoC - cleanup

### PR DESCRIPTION
**Pull request by przemyslawfierek-silvair**

### CODE REVIEW

File: mesh/amqp.c [hunk 1]
```diff
@@ -460,125 +558,138 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
...
-	l_free(amqp->exchange);
-	l_free(amqp->routing_key);
 	l_free(amqp);
 }
 
-const char *mesh_amqp_get_url(struct mesh_amqp *amqp)
+static void *get_ret_message(struct mesh_amqp *amqp)
 {
-	return amqp->url;
+	while (1) {
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c:618]: Ewwwwwww

Comment (przemyslawfierek-silvair) [reply mesh/amqp.c:618]: I've added a comment. Please read and say is it ok or not :)

File: mesh/amqp.c [hunk 2]
```diff
@@ -21,63 +21,98 @@
...
+	GET_EXCHANGE,
+	SET_ROUTING_KEY,
+	GET_ROUTING_KEY,
+	GET_READY_STATUS,
 	PUBLISH,
-	CLOSE,
+	STOP,
+};
+
+enum amqp_state_type {
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c]: Drop `_type` suffix

File: mesh/amqp.h [hunk 3]
```diff
@@ -21,15 +21,27 @@
...
+};
+
 struct mesh_amqp *mesh_amqp_new(void);
 void mesh_amqp_free(struct mesh_amqp *amqp);
 
 const char *mesh_amqp_get_url(struct mesh_amqp *amqp);
-bool mesh_amqp_set_url(struct mesh_amqp *amqp, const char *url);
+void mesh_amqp_set_url(struct mesh_amqp *amqp, const char *url);
+
 const char *mesh_amqp_get_exchange(struct mesh_amqp *amqp);
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.h]: If this is supposed to be `free()d` by the caller, then return a non-const pointer. This way you won't need to cast when calling `l_free()`

File: mesh/amqp.c [hunk 4]
```diff
@@ -460,125 +593,143 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
...
 	l_free(amqp);
 }
 
-const char *mesh_amqp_get_url(struct mesh_amqp *amqp)
+static void *get_ret_message(struct mesh_amqp *amqp)
 {
-	return amqp->url;
+	// ELL DBus API requires that the getters functions will be blocking.
+	// To fullfill this requirements there is needed construction that
+	// allows to wait for incoming value and iterate the main loop.
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c]: Wording: "ELL DBus API requires that getters are blocking, so keep iterating the loop while we wait for reply from the thread".

Also, don't use C++ style comments.

Another question: what if the thread never replies?

File: mesh/amqp.c [hunk 5]
```diff
@@ -21,63 +21,98 @@
...
 #include <pthread.h>
-#include <errno.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/timerfd.h>
 #include <unistd.h>
-#include <stdio.h>
+#include <string.h>
+
+static const char *NOT_CONFIGURED = "";
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c]: I don't like it, just use `""` literal in the code.

File: mesh/amqp.c [hunk 6]
```diff
@@ -460,125 +593,143 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
 void mesh_amqp_free(struct mesh_amqp *amqp)
 {
 	if (amqp->thread_started) {
-		mesh_amqp_close(amqp);
+		mesh_amqp_stop(amqp);
 		pthread_join(amqp->thread, NULL);
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c]: I'd rather return thread context from thread function and free it here, just to keep deallocation in the same "place" as allocation.

File: mesh/amqp.c (hunk 4 above)
Comment (przemyslawfierek-silvair) [reply mesh/amqp.c]: Everything will live, but every call to the not responding thread will cause that the call stack will grow up. IMO some timeout can be added here.

Comment (przemyslawfierek-silvair) [reply mesh/amqp.c]: I've added timeout

File: mesh/amqp.c [hunk 7]
```diff
@@ -460,125 +581,177 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
...
+	while (1) {
+		ret_msg = l_queue_pop_head(amqp->ret_queue);
+		if (ret_msg)
+			break;
 
-	strncpy(msg->connect.vhost, strlen(info.vhost) ? info.vhost : "/", sizeof(msg->connect.vhost) - 1);
-	strncpy(msg->connect.user, info.user, sizeof(msg->connect.user) - 1);
-	strncpy(msg->connect.pass, info.password, sizeof(msg->connect.pass) - 1);
+		if (expired) {
+			l_error("Timeout: return message hasn't came");
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c]: "amqp: command timed out"

**Review by michallowasrzechonek-silvair — APPROVED** [review]
Change log wording and LGTM, thanks!

**Pull request merged**